pymatgen/analysis/diffraction/tem.py

```diff
@@ -0,0 +1,520 @@
+        Returns:
+            dict from atomic symbol to another dict of hkl plane to factor
+        """
+        electron_scattering_factors = {}
+        x_ray_factors = self.x_ray_factors(structure, bragg_angles)
+        s2 = self.get_s2(bragg_angles)
+        atoms = structure.composition.elements
+        prefactor = sc.e / (16 * (np.pi ** 2) * sc.epsilon_0)
+        scattering_factors_for_atom = {}
+        for atom in atoms:
+            for plane in bragg_angles:
+                scattering_factor_curr = prefactor * (atom.Z - x_ray_factors[atom.symbol][plane]) / s2[plane]
+                scattering_factors_for_atom[plane] = scattering_factor_curr
+            electron_scattering_factors[atom.symbol] = scattering_factors_for_atom
+            scattering_factors_for_atom = {}
+        return electron_scattering_factors
+
+    def cell_scattering_factors(self, structure: Structure, bragg_angles: Dict[Tuple[int, int, int], float]) \
+            -> Dict[Tuple[int, int, int], int]:
+        """
+        Calculates the scattering factor for the whole cell.
+        Args:
+            structure (Structure): The input structure.
+            bragg_angles (dict of 3-tuple to float): The Bragg angles for each hkl plane.
+        Returns:
+            dict of hkl plane (3-tuple) to scattering factor
+        """
+        cell_scattering_factors = {}
+        electron_scattering_factors = self.electron_scattering_factors(structure, bragg_angles)
+        scattering_factor_curr = 0
+        for plane in bragg_angles:
+            for site in structure:
+                for sp, occu in site.species.items():
+                    g_dot_r = np.dot(np.array(plane), np.transpose(site.frac_coords))
+                    scattering_factor_curr += electron_scattering_factors[sp.symbol][plane] * np.exp(
+                        2j * np.pi * g_dot_r)
+            cell_scattering_factors[plane] = scattering_factor_curr
+            scattering_factor_curr = 0
+        return cell_scattering_factors
+
+    def cell_intensity(self, structure: Structure, bragg_angles: Dict[Tuple[int, int, int], float]) \
+            -> Dict[Tuple[int, int, int], float]:
+        """
+        Calculates cell intensity for each hkl plane. For simplicity's sake, take I = |F|**2.
+        Args:
+            structure (Structure): The input structure.
+            bragg_angles (dict of 3-tuple to float): The Bragg angles for each hkl plane.
+        Returns:
+            dict of hkl plane to cell intensity
+        """
+        csf = self.cell_scattering_factors(structure, bragg_angles)
+        plane = bragg_angles.keys()
+        csf_val = np.array(list(csf.values()))
+        cell_intensity_val = (csf_val * csf_val.conjugate()).real
+        cell_intensity = dict(zip(plane, cell_intensity_val))
+        return cell_intensity
+
+    def get_pattern(self, structure: Structure, scaled: bool = True, two_theta_range: tuple = (0, 90)) \
```

Comment:
type should be `Tuple[float, float]`?

Reply:
I am using hkl plane to a float here, so I think it is Tuple[int, int, int]?

Reply:
Sorry, I was referring to the two_theta range specifically in get_pattern

Reply:
Ah! Sorry, those arguments were left from previous implementation of the method, which were relevant to powder diffraction, and as I work on the code, I found them to be unnecessary. They should not have an effect on the code.